### PR TITLE
Optimize set_control_data

### PR DESCRIPTION
**Pull request by mkosunen**
I already merged this after throughly testing it (I thought). However, it seems that it breaks Non event-based IO's.
Try to run inverter sv sim with it. 

@chiplet

### CODE REVIEW

**Review by mkosunen — COMMENTED** [review]
I'll try first approach to use this sorted dict only for event data.

**Comment by mkosunen** [comment]
@chiplet 
I think now it works. Can you confirm?

File: rtl/rtl_iofile.py [hunk 1]
```diff
@@ -267,26 +272,72 @@ def set_control_data(self,**kwargs):
...
+                    transposed[i,j] = transposed[i, j-1]
+        io_array = np.transpose(transposed).astype(int)
+
+        return io_array
+
+    # Setter - Takes a numpy array and converts it to the diff-based SortedDict
+    @Data.setter
+    def Data(self, value):
+        # convert value to equivalent SortedDict representation
+        self.DictData = sc.SortedDict()
```
[the remark this answers is not in the document]
Comment (chiplet) [reply rtl/rtl_iofile.py]: No it doesn't, this was written under the assumption that the first column is the time column, but I see you've already fixed it by explicitly checking for the event type.

**Comment by chiplet** [comment]
@mkosunen 
JTAG programming in simulation still works after these changes. 👍 

When running the sv model for the inv_sim entity I can see a delay of one cycle. Should this happen? This behaviour doesn't seem to depend on the changes in this PR though as reverting to the preceding `rtl` commit still shows the one cycle delay.

Screenshot 2021-08-11 at 12 13 27

**Review by chiplet — APPROVED** [review]

**Comment by mkosunen** [comment]
Thatis controlled with the 'latency' vector in main guard. For rtl simulations, latency should be 1 as file IO is syncronous to clock. This is ugly, but most of the designs are syncronous anyway. Asychronous inverter is an exception.

**Comment by mkosunen** [comment]
I'l' merge this anyway as I see it is clear and good. Let's handle that spice feature independently.

**Pull request merged**